File: mainActivity/src/main/java/com/sierrawireless/avphone/task/IApplicationClient.java
package com.sierrawireless.avphone.task;

import java.io.IOException;
import java.util.ArrayList;

import net.airvantage.model.AirVantageException;
import net.airvantage.model.Application;
import net.airvantage.model.AvSystem;

import com.sierrawireless.avphone.model.AvPhoneObjectData;

public interface IApplicationClient {

    Application ensureApplicationExists() throws IOException, AirVantageException;

    void setApplicationData(String applicationUid, ArrayList<AvPhoneObjectData> customData, String object) throws IOException, AirVantageException;

    Application createApplication() throws IOException, AirVantageException;

    void setApplicationCommunication(String applicationUid) throws IOException, AirVantageException;

    void addApplication(AvSystem system, Application application) throws IOException, AirVantageException;

}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/IApplicationClient.kt
package com.sierrawireless.avphone.task

import com.sierrawireless.avphone.model.AvPhoneObject
import net.airvantage.model.AirVantageException
import net.airvantage.model.Application
import net.airvantage.model.AvSystem
import java.io.IOException

interface IApplicationClient {

    @Throws(IOException::class, AirVantageException::class)
    fun ensureApplicationExists(phoneName:String): Application

    @Throws(IOException::class, AirVantageException::class)
    fun setApplicationData(applicationUid: String, obj: AvPhoneObject)

    @Throws(IOException::class, AirVantageException::class)
    fun createApplication(phoneName: String): Application

    @Throws(IOException::class, AirVantageException::class)
    fun setApplicationCommunication(applicationUid: String)

    @Throws(IOException::class, AirVantageException::class)
    fun addApplication(system: AvSystem, application: Application)
}